x/evm/keeper/keeper.go

```diff
@@ -229,50 +229,124 @@ func (k Keeper) ResetRefundTransient(ctx sdk.Context) {
 // GetAllTxLogs return all the transaction logs from the store.
 func (k Keeper) GetAllTxLogs(ctx sdk.Context) []types.TransactionLogs {
 	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixLogs)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixTxLogNum)
 	defer iterator.Close()
 
 	txsLogs := []types.TransactionLogs{}
 	for ; iterator.Valid(); iterator.Next() {
-		var txLog types.TransactionLogs
-		k.cdc.MustUnmarshal(iterator.Value(), &txLog)
+		logNum := new(big.Int).SetBytes(iterator.Value())
+		txHash := iterator.Key()
+
+		var txlogs types.TransactionLogs
+		txlogs.Hash = common.Bytes2Hex(txHash)
+
+		s := prefix.NewStore(k.Ctx().KVStore(k.storeKey), types.KeyPrefixLogs)
+		for i := int64(0); i < logNum.Int64(); i++ {
+			var key []byte
+			key = append(key, txHash...)
+			key = append(key, new(big.Int).SetInt64(i).Bytes()...)
+
+			logbz := s.Get(key)
+			if logbz == nil {
+				// TODO: error handle?
+				continue
+			}
+
+			var txLog types.Log
+			k.cdc.MustUnmarshal(logbz, &txLog)
+			txlogs.Logs = append(txlogs.Logs, &txLog)
+		}
 
 		// add a new entry
-		txsLogs = append(txsLogs, txLog)
+		txsLogs = append(txsLogs, txlogs)
 	}
 	return txsLogs
 }
 
 // GetLogs returns the current logs for a given transaction hash from the KVStore.
 // This function returns an empty, non-nil slice if no logs are found.
 func (k Keeper) GetTxLogs(txHash common.Hash) []*ethtypes.Log {
-	store := prefix.NewStore(k.Ctx().KVStore(k.storeKey), types.KeyPrefixLogs)
+	store := prefix.NewStore(k.Ctx().KVStore(k.storeKey), types.KeyPrefixTxLogNum)
 
 	bz := store.Get(txHash.Bytes())
-	if len(bz) == 0 {
+	if bz == nil {
+		return []*ethtypes.Log{}
+	}
+
+	logNum := new(big.Int).SetBytes(bz)
```

Comment:
ditto, why big.Int?

Reply:
you mean int64?

Reply:
https://github.com/tharsis/ethermint/pull/461#discussion_r693009503